File: backend/app/schemas/vehicule.py
from pydantic import BaseModel
from datetime import date
from typing import Optional
from app.schemas.client import ClientResponse


class VehiculeBase(BaseModel):
    matricule: str
    marque: Optional[str] = None
    model: Optional[str] = None
    date_mise_en_circulation: Optional[date] = None
    valeur_venale: Optional[float] = None
    puissance_fiscale: Optional[float] = None
    fk_client_id: Optional[int] = None


class VehiculeCreate(VehiculeBase):
    matricule: str
    fk_client_id: int


class VehiculeUpdate(BaseModel):
    matricule: Optional[str] = None
    marque: Optional[str] = None
    model: Optional[str] = None
    date_mise_en_circulation: Optional[date] = None
    valeur_venale: Optional[float] = None
    puissance_fiscale: Optional[float] = None
    fk_client_id: Optional[int] = None


class VehiculeResponse(VehiculeBase):
    id: int
    client: Optional[ClientResponse] = None

    class Config:
        from_attributes = True

File: frontend/src/types/vehicule.ts
import { Client } from "./client";

export interface Vehicule {
  id: number;
  matricule: string;
  marque: string;
  model: string;
  date_mise_en_circulation: string; 
  valeur_venale: number;
  puissance_fiscale: number;
  fk_client_id?: number;
  client?: Client;
}

export interface VehiculeCreate {
  matricule: string;
  marque: string;
  model: string;
  date_mise_en_circulation: string; 
  valeur_venale: number;
  puissance_fiscale: number;
  fk_client_id?: number;
}
export interface VehiculeUpdate {
  matricule: string;
  marque: string;
  model: string;
  date_mise_en_circulation: string; 
  valeur_venale: number;
  puissance_fiscale: number;
  fk_client_id?: number;
}